src/ShipElement.java

```diff
@@ -0,0 +1,16 @@
+
```

Comment:
the Ship class already implements the observer pattern. Not sure what you were trying to accomplish here.